### PR TITLE
[NCL-7583]  - Send the PNC BuildConfig ID to Indy when clearing temporary builds

### PR DESCRIPTION
**Pull request by vibe13**

### CODE REVIEW

**Comment by vibe13** [comment]
@michalovjan @pkocandr Hi! So the PR build fails because there are a few SNAPSHOT dependencies which I need to check more closely (one more time). Anyway, would you mind giving a look please? Thanks!

File: src/main/java/org/jboss/pnc/cleaner/archiveservice/ArchivesCleanerImpl.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    @Inject
+    Config config;
+
+    @Inject
+    ManagedExecutor executor;
+
+    @Inject
+    java.net.http.HttpClient httpClient;
+
+    private String DELETE_ARCHIVE_URL;
```
Comment (michalovjan) [inline src/main/java/org/jboss/pnc/cleaner/archiveservice/ArchivesCleanerImpl.java:77]: Is there a reason for not using ConfigProperty from MP Config?

File: src/main/java/org/jboss/pnc/cleaner/archiveservice/ArchivesCleanerImpl.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+                .with(executor)
+                .getStageAsync(
+                        () -> httpClient.sendAsync(request, HttpResponse.BodyHandlers.ofString())
+                                .thenApply(validateResponse()))
+                .join();
+    }
+
+    private Function<HttpResponse<String>, HttpResponse<String>> validateResponse() {
+        return response -> {
+            if (response.statusCode() >= 200 && response.statusCode() < 300) {
```
Comment (michalovjan) [inline src/main/java/org/jboss/pnc/cleaner/archiveservice/ArchivesCleanerImpl.java:164]: are redirects valid?

File: src/main/java/org/jboss/pnc/cleaner/archiveservice/ArchivesCleanerImpl.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+                return response;
+            } else {
+                throw new FailedResponseException("Response status code: " + response.statusCode());
```
Comment (michalovjan) [inline src/main/java/org/jboss/pnc/cleaner/archiveservice/ArchivesCleanerImpl.java:167]: Are we interested in message? Can we also log.error it with message?

File: src/main/java/org/jboss/pnc/cleaner/logverifier/BuildLogVerifier.java [hunk 4]
```diff
@@ -60,6 +60,7 @@ public class BuildLogVerifier {
     private static final String className = BuildLogVerifier.class.getName();
 
     private final Logger logger = LoggerFactory.getLogger(BuildLogVerifier.class);
+    private final Integer DEFAULT_BIFROST_BATCH_SIZE = 1;
```
Comment (michalovjan) [inline src/main/java/org/jboss/pnc/cleaner/logverifier/BuildLogVerifier.java:63]: BatchSize with value 1 means that, 1 batch has size 1. Translated to 1 line per DB request or in other way 100000DB requests for 100000 log lines. Could you put it to something like 10000?

**Review by michalovjan — CHANGES_REQUESTED** [review]

File: src/main/java/org/jboss/pnc/cleaner/archiveservice/ArchivesCleanerImpl.java (hunk 1 above)
Comment (vibe13) [reply src/main/java/org/jboss/pnc/cleaner/archiveservice/ArchivesCleanerImpl.java:77]: no, it's some leftover from previous code which I haven't cleaned up. Fixing :-)

File: src/main/java/org/jboss/pnc/cleaner/archiveservice/ArchivesCleanerImpl.java (hunk 2 above)
Comment (vibe13) [reply src/main/java/org/jboss/pnc/cleaner/archiveservice/ArchivesCleanerImpl.java:164]: good point, I have checked the swagger docs (now that they are available), and 202 is returned. fixing.

File: src/main/java/org/jboss/pnc/cleaner/archiveservice/ArchivesCleanerImpl.java (hunk 3 above)
Comment (vibe13) [reply src/main/java/org/jboss/pnc/cleaner/archiveservice/ArchivesCleanerImpl.java:167]: adding, yes might be interesting to know more

File: src/main/java/org/jboss/pnc/cleaner/logverifier/BuildLogVerifier.java (hunk 4 above)
Comment (vibe13) [reply src/main/java/org/jboss/pnc/cleaner/logverifier/BuildLogVerifier.java:63]: sure, I was waiting for your feedback especially about this! setting to 10000 then :-)

File: src/main/java/org/jboss/pnc/cleaner/archiveservice/ArchivesCleanerImpl.java (hunk 2 above)
Comment (pkocandr) [reply src/main/java/org/jboss/pnc/cleaner/archiveservice/ArchivesCleanerImpl.java:157]: I would not either. If there is a redirect, we need to update the configuration.

**Review by michalovjan — APPROVED** [review]

**Review by pkocandr — APPROVED** [review]
Looks good! Sorry that it took so long.

**Comment by vibe13** [comment]
thanks guys! :-) merging

**Pull request merged**